x/authz/module/module.go

```diff
@@ -167,6 +174,37 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 	return []abci.ValidatorUpdate{}
 }
 
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(
+			provideModuleBasic,
+			provideModule,
+		),
+	)
+}
+
+func provideModuleBasic() runtime.AppModuleBasicWrapper {
+	return runtime.WrapAppModuleBasic(AppModuleBasic{})
+}
+
+type authzInputs struct {
+	depinject.In
+
+	KVStoreKey       *store.KVStoreKey
+	Cdc              codec.Codec
+	AccountKeeper    authz.AccountKeeper `key:"cosmos.auth.v1.AccountKeeper"`
+	BankKeeper       authz.BankKeeper    `key:"cosmos.bank.v1.Keeper"`
+	Registry         cdctypes.InterfaceRegistry
+	MsgServiceRouter *baseapp.MsgServiceRouter
+}
+
+func provideModule(in authzInputs) (keeper.Keeper, runtime.AppModuleWrapper) {
+	k := keeper.NewKeeper(in.KVStoreKey, in.Cdc, in.MsgServiceRouter, in.AccountKeeper)
```

Comment:
to initiate the authz keeper `msgServiceRouter` is required, anyone know how do I get `app.MsgServiceRouter()` here?

Reply:
We can have the `runtime` module provide it